src/components/MultiHopTrade/components/FiatRamps/FiatRampTrade.tsx

```diff
@@ -42,127 +68,299 @@ type RampRoutesProps = {
   isCompact?: boolean
   isStandalone?: boolean
   onChangeTab: (newTab: TradeInputTab) => void
-  type: 'buy' | 'sell'
+  direction: FiatRampAction
 }
 
-const RampRoutes = memo(({ onChangeTab, type }: RampRoutesProps) => {
+const RampRoutes = memo(({ onChangeTab, direction }: RampRoutesProps) => {
   const tradeInputRef = useRef<HTMLDivElement | null>(null)
+  const dispatch = useAppDispatch()
+
+  const sellAsset = useAppSelector(selectInputSellAsset)
+  const buyAsset = useAppSelector(selectInputBuyAsset)
+  const sellAmountCryptoPrecision = useAppSelector(selectInputSellAmountCryptoPrecision)
+  const hasUserEnteredAmount = useAppSelector(selectHasUserEnteredAmount)
+  const maybeSellFiatCurrency = useAppSelector(selectSellFiatCurrency)
+  const maybeBuyFiatCurrency = useAppSelector(selectBuyFiatCurrency)
+  const sellFiatAmount = useAppSelector(selectSellFiatAmount)
+  const selectedQuote = useAppSelector(selectSelectedFiatRampQuote)
+  const buyAccountId = useAppSelector(selectBuyAccountId)
+
+  const sellFiatCurrency =
+    maybeSellFiatCurrency ?? fiatCurrencyObjectsByCode[FiatCurrencyTypeEnum.USD]
+  const buyFiatCurrency =
+    maybeBuyFiatCurrency ?? fiatCurrencyObjectsByCode[FiatCurrencyTypeEnum.USD]
```

Comment:
q: dis expected for both to fallback to USD? 

q2: Also, do we really want a fallback and shouldn't we not assume *maybe* but have a proper default for `sellFiatCurrency` / `buyFiatCurrency`?